src/shogun/classifier/LDA.cpp

```diff
@@ -95,45 +113,49 @@ bool CLDA::train_machine(CFeatures *data)
 			classidx_pos[num_pos++]=i;
 	}
 
-	w=SGVector<float64_t>(num_feat);
-	w.zero();
-	MatrixXd fmatrix=Map<MatrixXd>(feature_matrix.matrix, num_feat, num_vec);
-	VectorXd mean_neg(num_feat);
-	mean_neg=VectorXd::Zero(num_feat);
-	VectorXd mean_pos(num_feat);
-	mean_pos=VectorXd::Zero(num_feat);
+	SGVector<ST> w_st(num_feat);
+	w_st.zero();
+	typename SGMatrix<ST>::EigenMatrixXt fmatrix = typename SGMatrix<ST>::EigenMatrixXtMap(feature_matrix.matrix, num_feat, num_vec);
+	typename SGVector<ST>::EigenVectorXt mean_neg(num_feat);
+	mean_neg.setZero();
+	typename SGVector<ST>::EigenVectorXt mean_pos(num_feat);
+	mean_pos.setZero();
 
 	//mean neg
-	for(i=0; i<num_neg; i++)
-		mean_neg+=fmatrix.col(classidx_neg[i]);
-	mean_neg/=(float64_t)num_neg;
+	for(i=0; i<num_neg; i++){
+		mean_neg+= fmatrix.col(classidx_neg[i]);
+	}
+
+	mean_neg/= (ST) num_neg;
```

Comment:
Style question: should I also do this for anything of the form [operator]= (e.g. mean_neg+=)?